lib/cloud-metadata/gcp.js

```diff
@@ -5,9 +5,13 @@
  */
 
 'use strict';
+
 const URL = require('url').URL;
+const JSONBigInt = require('json-bigint');
```

Comment:
Reviewer note: This is the same lib that OTel is indirectly using. They use the Google-provided "gcp-metadata" package. I didn't want to use that because it adds yet another HTTP lib that we don't particularly need (gaxios, a wrapper around node-fetch that provides an interface like `axios`).  

The "bignumber.js" dep (used by `json-bigint`) adds about 1MB to the install size, I think, which is unfortunate. We *could* get into shenanigans like using a regex to parse out the "id". :)

Reply:
Checking the [metadata docs](https://cloud.google.com/compute/docs/metadata/default-metadata-values#instance-attributes-metadata) and also the JSON fixture you added I think we can be sure that there are only two numeric values in the response: `numericProjectId` and `id` (within instance)

```json
{
  "instance": {
    "attributes": {},
    "id": 5737554347302044216,
    ...
  },
  "project": {
    "attributes": {
      ...
    },
    "numericProjectId": 523926462582,
    "projectId": "acme-eng"
    ...
  }
}
```

Since it is so specific I think we can opt-in to a custom solution to capture that info. You mentioned `RegExp` and maybe is the way to go using capture groups with `String.match`. We can capture the numeric values with something similar to the snippet below and use `JSON.parse` for the rest

```js
const respBody = '...' // The response text with all the metadata

const instanceIdRegExp = /"id":(\d+),/;
const projectIdRegExp = /"numericProjectId":(\d+),/;

const instanceIdMatch = respBody.match(instanceIdRegExp)
const projectIdMatch = respBody.match(projectIdRegExp)

// The match is already an String so no need to convert
const instanceId = instanceIdMatch && instanceMatch[1];
const projectId = projectIdMatch && projectIdMatch[1];

```

NOTE: regexes should be reviewed since they do not cover the case of being the last property of the object

Reply:
David and I discussed. The added "bignumber.js" dep *is* a bit of size, but still relatively low compared to a number of other deps, so we think it is better to have the clearer implementation and not get into possibly brittle regex usage just to save some install size.